### PR TITLE
util/time: add a custom Instant

### PR DESCRIPTION
**Pull request by overvenus**
This new `Instant` is compatible with `std::time::Instant`, and has some extra features, e.g., `now_coarse` is much faster than `now`, it should be useful for performance-critical subroutines.

### CODE REVIEW

File: src/util/time.rs [hunk 1]
```diff
@@ -171,16 +191,114 @@ mod inner {
...
+        };
+        let earlier = match earlier {
+            Instant::Monotonic(t) |
+            Instant::MonotonicCoarse(t) => t,
+        };
+        elapsed_duration(later, earlier)
+    }
+}
+
+impl Add<Duration> for Instant {
```
Comment (BusyJay) [inline src/util/time.rs]: I don't think this is necessary.

File: src/util/time.rs [hunk 2]
```diff
@@ -171,16 +191,114 @@ mod inner {
...
+/// for providing extra features.
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug)]
+pub enum Instant {
+    Monotonic(Timespec),
+    MonotonicCoarse(Timespec),
+}
+
+impl Instant {
+    pub fn now() -> Instant {
+        Instant::Monotonic(raw_now())
```
Comment (siddontang) [inline src/util/time.rs]: why raw here? monotonic is different with monotonic raw.

File: src/util/time.rs (hunk 1 above)
Comment (overvenus) [inline src/util/time.rs]: Just for being compatible with `std::time::Instant`.

Comment (BusyJay) [inline src/util/time.rs]: This is a util for TiKV, not a standard alone crate, please keep it clean and concise.

Comment (overvenus) [inline src/util/time.rs]: It is **necessary**. https://github.com/pingcap/tikv/blob/13a8b89a94c0bda9f812df003d0d3a4455801034/src/coprocessor/endpoint.rs#L131-L134

Comment (BusyJay) [inline src/util/time.rs]: Got it.

**Comment by siddontang** [comment]
LGTM
PTAL @BusyJay

File: src/util/time.rs [hunk 3]
```diff
@@ -170,17 +199,126 @@ mod inner {
...
+    }
+
+    fn get_time(clock: libc::clock_t) -> Timespec {
         let mut t = libc::timespec {
             tv_sec: 0,
             tv_nsec: 0,
         };
-        let res = unsafe { libc::clock_gettime(libc::CLOCK_MONOTONIC_RAW, &mut t) };
-        if res != 0 {
+        let errno = unsafe { libc::clock_gettime(clock as _, &mut t) };
```
Comment (BusyJay) [inline src/util/time.rs]: Why cast?

Comment (overvenus) [inline src/util/time.rs]: Rustc blames: 

```
error[E0308]: mismatched types
   --> src/util/time.rs:219:50
    |
219 |         let errno = unsafe { libc::clock_gettime(clock, &mut t) };
    |                                                  ^^^^^ expected i32, found i64

```

Comment (overvenus) [inline src/util/time.rs]: Oops, I made a mistake. The type should be `libc::clockid_t`, sorry!

**Pull request merged**